packages/@ionic/cli/src/lib/project/angular/generate.ts

```diff
@@ -98,14 +98,15 @@ To test a generator before file modifications are made, use the ${input('--dry-r
     }
   }
 
-  createOptionsFromCommandLine(inputs: CommandLineInputs, options: CommandLineOptions): AngularGenerateOptions {
-    const baseOptions = super.createOptionsFromCommandLine(inputs, options);
+  async createOptionsFromCommandLine(inputs: CommandLineInputs, options: CommandLineOptions): Promise<AngularGenerateOptions> {
+    const baseOptions: GenerateOptions = await super.createOptionsFromCommandLine(inputs, options);
+    const project = options['project'] ? String(options['project']) : await this.e.project.getWorkspaceDefaultProject();
 
     // TODO: this is a little gross, is there a better way to pass in all the
     // options that the command got?
     return {
       ...lodash.omit(options, '_', '--', ...GLOBAL_OPTIONS.map(opt => opt.name)),
-      project: options['project'],
+      project,
```

Comment:
The `ionic generate` angular command was the only one that didn't provide the default project name fallback (`app`). I've added it here and added a few tests.